packages/shared/src/hooks/useUserMention.ts

```diff
@@ -0,0 +1,205 @@
+import { useQuery, useQueryClient } from 'react-query';
```

Comment:
This file can use some more refactoring and better naming.
Some of the functions here are hard to follow and understand because there's quite a lot of complex logic here